### PR TITLE
feat: Cascader support customize clear button

### PR DESCRIPTION
**Pull request by kiner-tang**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [X] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 🔗 Related issue link

- https://github.com/ant-design/ant-design/discussions/43582
- https://github.com/ant-design/ant-design/issues/43495

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Cascader support customize clear button |
| 🇨🇳 Chinese | 级联选择组件支持自定义清除按钮  |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [X] Doc is updated/provided or not needed
- [X] Demo is updated/provided or not needed
- [X] TypeScript definition is updated/provided or not needed
- [X] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at e420bd4</samp>

The `allowClear` property of the `Cascader` component was enhanced to support custom clear icons and changed to default to `false`. The documentation files `index.en-US.md` and `index.zh-CN.md` were updated to reflect this change.

### <samp>🤖 Generated by Copilot at e420bd4</samp>

* Updated the `allowClear` property of the Cascader component to accept an object with a `clearIcon` property, which allows customizing the clear icon. Changed the default value from `true` to `false`. ([link](https://github.com/ant-design/ant-design/pull/43776/files?diff=unified&w=0#diff-5ced7e6e1259091e502d7b0512332fdeb05319823a37c5ae3fa7ffb1ccf8aa50L49-R51), [link](https://github.com/ant-design/ant-design/pull/43776/files?diff=unified&w=0#diff-64c67e20c18f62d937b9af3bc101a2a1b9cd92df3ac46d93a8e252683297b5f3L50-R52))

### CODE REVIEW

File: components/cascader/index.en-US.md [hunk 1]
```diff
@@ -46,10 +46,9 @@ Cascade selection box.
 
 | Property | Description | Type | Default | Version |
 | --- | --- | --- | --- | --- |
-| allowClear | Whether allow clear | boolean | true |  |
+| allowClear | Show clear button, 5.8.0: `{ clearIcon?: ReactNode }` | boolean \| { clearIcon?: ReactNode } | false |  |
```
Comment (zombieJ) [inline components/cascader/index.en-US.md:49]: 版本信息挪到 version 里~

**Review by zombieJ — APPROVED** [review]

File: components/cascader/index.en-US.md (hunk 1 above)
Comment (kiner-tang) [reply components/cascader/index.en-US.md:49]: 主要是这个信息有点长，放在 version 宽度不够看起来比较丑，以前遇到这种情况都是放到描述里面的

Comment (afc163) [reply components/cascader/index.en-US.md:49]: 还是统一放后面比较好，丑点没关系。描述里可以继续放文字说明，比如 `5.8.0 后可扩展为 object 形式`

Comment (kiner-tang) [reply components/cascader/index.en-US.md:49]: 好的，那我调整一下

**Pull request merged**